### PR TITLE
Sc/ddan 28 resource monitor

### PR DESCRIPTION
**Pull request by Sangmin0728**
### PURPOSE OF PR
* Simple summary of this PR
test code for Resource Monitor

### What type of PR is it?
* feature

### JIRA ISSUE
* https://ddangcoonfire.atlassian.net/browse/DDAN-28

### HAVE YOU CHECKED ...
* [v] ADD JIRA ID ON PR TITLE (ex. [DDAN-1234] PR TITLE)
* [v] pre-commit
* [v] Branch Format ( type/issue-NUM ) (ex. refactoring/DDAN-1234 )

### Explanation
( Adding screenshot would be great help )

### CODE REVIEW

File: experiment_scheduler/master/master.py [hunk 1]
```diff
@@ -181,23 +182,26 @@ def get_available_task_managers(self):
         # return available_task_managers
         return [self.task_managers_address[0]]
 
-    def execute_task(self, task_manager):
+    async def execute_task(self, task_manager):
```
Comment (ptaejoon) [inline experiment_scheduler/master/master.py:185]: why does execute_task have to be async function?

File: experiment_scheduler/master/master.py [hunk 2]
```diff
@@ -181,23 +182,26 @@ def get_available_task_managers(self):
...
-        self.master_pipes[task_manager].send(
-            [
-                "run_task",
-                gpu_idx,
-                prior_task.command,
-                prior_task.name,
-                dict(prior_task.task_env),
-            ]
-        )
+        while (True):
```
Comment (ptaejoon) [inline experiment_scheduler/master/master.py:191]: why infinite loop is needed?  _execute_command which calls execute_task also use infinite loop.

**Review by ptaejoon — COMMENTED** [review]
Resource Monitor must notify available task_manager using get_available_task_managers

File: experiment_scheduler/master/master.py [hunk 3]
```diff
@@ -52,15 +54,18 @@ def _execute_command(self, interval=1):
...
-                    task_manager = available_task_managers.pop(0)
-                    self.execute_task(task_manager)
-            # [TODO] master pipe recv logic required here
-            for task_manager, pipe in self.master_pipes.items():
-                if pipe.poll():
-                    print(pipe.recv())  # need change later
+            gpu_idx = responser.get_max_free_gpu()
+            if gpu_idx != -1:
+                available_task_managers = self.get_available_task_managers()
+                self.resource_monitor.queued_gpu_idx.append(gpu_idx)
```
Comment (ptaejoon) [inline experiment_scheduler/master/master.py:60]: GPU id popped from queued_gpu_idx must ensure the gpu is in same available_task_manager.

Comment (ptaejoon) [reply experiment_scheduler/master/master.py:60]: I recommend to group task_manger_address (from get_availabale_task_managers) and  gpu index into a tuple, returned from get_available_task_manager.

**Comment by Armstrong-SuHwan** [comment]
i checked changes available gpu idx was in master's execute command

**Comment by Armstrong-SuHwan** [comment]
resolve merge conflict.

**Comment by Armstrong-SuHwan** [comment]
i checked changes about resource monitor

**Review by Armstrong-SuHwan — APPROVED** [review]
i checked all changes about resource monitor

**Pull request merged**